src/dmd/utils.d

```diff
@@ -210,6 +140,60 @@ void escapePath(OutBuffer* buf, const(char)* fname)
     }
 }
 
+/**
+ * Takes a path, and make it compatible with Makefile format.
+ *
+ * That is, turns
+ *  - '\' into '/'
+ *  - '$' into "$$"
+ *  - ' ' into "\ "
+ *  - '#' into "\#"
+ *  - ':' into "\:"
+ *
```

Comment:
I will document that it output in GNU/unix compatible format.
Do you see any possibility that this would be actually parsed by nmake?
AFAICT only tools like Ninja, and may be Dub some day will actually read the output of this.
From [Ninja source](https://github.com/ninja-build/ninja/blob/68f971460bb21abd59dfe6253e35a55c3e1cfc64/src/depfile_parser.in.cc#L81), it seems to expect the unix format.

Reply:
> Do you see any possibility that this would be actually parsed by nmake?

No. I just tried `\#`, `\:` and `\ `, they caused different kind of errors. Usually these are rather special cases, but ':' might show up with absolute import paths.

> AFAICT only tools like Ninja, and may be Dub some day will actually read the output of this.

It's advertised as output for make, though ;-)
I looked at a random ninja file and found this line:
```
build CMakeFiles\LDCShared.dir\gen\aa.cpp.obj: CXX_COMPILER__LDCShared C$:\s\d\ldc\ldc\gen\aa.cpp || cmake_object_order_depends_target_LDCShared
```
Note the `C$:`. Maybe escaping works differently on Windows for ninja, too?

Reply:
This looks more like a Ninja rules file (written by CMake or meson) than a make dependency file (as e.g. gcc would emit and ninja parse).
Ninja rules syntax clearly document `$` as the only escape character. See https://ninja-build.org/manual.html#ref_lexer

Reply:
Ah, ok. On Windows I didn't find any make style dependency files for projects built with ninja (all cmake generated), but a .ninja_deps file that has a binary format and is used as a database (as described here: https://ninja-build.org/manual.html#_deps ).

Reply:
It's right that `:` will be very common on windows.
It deserve proper testing. It is likely that not escaping at all would also work fine.